docs/getting_started.md

```diff
@@ -104,13 +104,23 @@ get_things_name = store.inject(get_things_name)
 print(get_things_name())  # prints "Thing"
 ```
 
+!!!note "multiple providers"
+    In the case that more one provider has been registered for a given type. The store will
+    iterate through all providers registered for the required type (ordered by [weight](#weights-and-provider-priority)), stopping at the first one that returns an object that
+    is not `None` .
+
+    Providers should return `None` if it is not able to provide the requested object as
+    this allows `in-n-out` to continue iterating through any other registered providers.
+
 !!!tip "decorators"
     As with registration functions, we can use `Store.inject` as a decorator:
 
     ```python
     @store.inject
     def get_things_name(thing: Thing) -> str:
-        return thing.name
+        if hasattr(thing, name):
+            return thing.name
+        return None
```

Comment:
if we're going to add this, then the return type annotation should be changed to `str | None`.   But I also feel like it's a bit contrived here... since `Thing` *always* has an attribute `name`... so it's not a great example of a processor that may or may not return a value (like, for example `napari.get_current_viewer`).  So, if you think an example of this is needed (in addition to the text above in the "multiple providers" note) then let's add a new section with a new provider

Reply:
Yeah examples are hard. I don't think it's needed, the doc should be enough